SQL/New_patches/2022-11-24-electrode-coord-system.sql

```diff
@@ -0,0 +1,194 @@
+  `Name`                            VARCHAR(20)           NOT NULL UNIQUE,
+  PRIMARY KEY (`PhysiologicalCoordSystemTypeID`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8;
+
+-- insert
+INSERT INTO physiological_coord_system_type
+  (`Name`)
+  VALUES
+  ('Not registered'),
+  ('MEG'),
+  ('EEG'),
+  ('iEEG'),
+  ('Fiducials'),
+  ('AnatomicalLandmark'),
+  ('HeadCoil'),
+  ('DigitizedHeadPoints');
+
+
+-- -----------------------------------------------------
+-- ADDED
+-- Table `physiological_coord_system_name`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `physiological_coord_system_name` (
+  `PhysiologicalCoordSystemNameID`  INT(10)     UNSIGNED  NOT NULL AUTO_INCREMENT,
+  `Name`                            VARCHAR(25)           NOT NULL UNIQUE,
+  `Orientation`                     VARCHAR(5)            NULL,
+  `Origin`                          VARCHAR(50)           NULL,
+  PRIMARY KEY (`PhysiologicalCoordSystemNameID`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8;
+
+-- insert
+INSERT INTO physiological_coord_system_name
+  (`Name`, `Orientation`, `Origin`)
+  VALUES
+  ('Not registered', NULL, NULL),
+  ('ACPC', 'RAS', 'anterior commissure'),
+  ('Allen Institute', 'RAS', 'Bregma point'),
+  ('Analyze', 'LAS', ''),
+  ('BTi/4D', 'ALS', 'between the ears'),
+  ('CTF MRI', 'ALS', 'between the ears'),
+  ('CTF gradiometer', 'ALS', 'between the ears'),
+  ('CapTrak', 'RAS', 'approximately between the ears'),
+  ('Chieti ITAB', 'RAS', 'between the ears'),
+  ('DICOM', 'LPS', 'centre of MRI gradient coil'),
+  ('EEGLAB', 'ALS', 'between the ears'),
+  ('FreeSurfer', 'RAS',	'center of isotropic 1mm 256x256x256 volume'),
+  ('MNI', 'RAS', 'anterior commissure'),
+  ('NIfTI', 'RAS', ''),
+  ('Neuromag/Elekta/Megin', 'RAS', 'between the ears'),
+  ('Paxinos-Franklin', 'RSP', 'Bregma point'),
+  ('Scanner RAS (scanras)', 'RAS', 'scanner origin'),
+  ('Talairach-Tournoux', 'RAS', 'anterior commissure'),
+  ('Yokogawa', 'ALS', 'center of device');
+
+-- -----------------------------------------------------
+-- ADDED
+-- Table `physiological_coord_unit`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `physiological_coord_system_unit` (
+  `PhysiologicalCoordSystemUnitID`  INT(10)     UNSIGNED  NOT NULL AUTO_INCREMENT,
+  `Name`                            VARCHAR(20)           NOT NULL UNIQUE,
+  `Symbol`                          VARCHAR(5)            NULL UNIQUE,
+  PRIMARY KEY (`PhysiologicalCoordSystemUnitID`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8;
+
+INSERT INTO physiological_coord_system_unit
+  (`Name`, `Symbol`)
+  VALUES
+  ('Not registered', NULL),
+  ('Millimeter', 'mm'),
+  ('Centimeter', 'cm'),
+  ('Meter', 'm');
+
+-- -----------------------------------------------------
+-- ADDED
+-- Table `physiological_coord_system`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `physiological_coord_system` (
```

Comment:
I think the landmarks are not always NAS/LPA/RPA, and we can also have HeadCoil/DigitizedHeadPoints as well. Maybe a more generalized approach would be to store head points in a separate table physiological_head_point:

`HeadPointID`               INT(10)     UNSIGNED  NOT NULL AUTO_INCREMENT,
`PhysiologicalCoordSystemID` INT(10)     UNSIGNED  NOT NULL,
`Name`                          VARCHAR(50)     NOT NULL,
`X`                                 DECIMAL(12,6)    NOT NULL,
`Y`                                 DECIMAL(12,6)    NOT NULL,
`Z`                                 DECIMAL(12,6)    NOT NULL

With FOREIGN KEY (`PhysiologicalCoordSystemID`)
REFERENCES `physiological_coord_system` (`PhysiologicalCoordSystemID`)

or anything equivalent.